File: app/src/main/java/com/me/noteapp/fragment/NoteEditFragment.java
package com.me.noteapp.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.TextView;

import com.me.noteapp.NotesApplication;
import com.me.noteapp.R;
import com.me.noteapp.dialog.DeleteDialogFragment;
import com.me.noteapp.entity.Item;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

public class NoteEditFragment extends Fragment {
    private EditText editNoteView;
    private boolean VIEW_FLAG = false;
    private Item mItem;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        mItem = (Item) getArguments().getSerializable("note");
        if (VIEW_FLAG) return inflateViewFrag(inflater, container);
        setHasOptionsMenu(true);
        return inflateEditFrag(inflater, container);
    }

    private View inflateViewFrag(LayoutInflater inflater, ViewGroup container) {
        View v = inflater.inflate(R.layout.fragment_note_view, container, false);
        ((TextView) v.findViewById(R.id.noteView)).setText(mItem.getContent());
        return v;
    }

    private View inflateEditFrag(LayoutInflater inflater, ViewGroup container) {
        View v = inflater.inflate(R.layout.fragment_note_edit, container, false);
        editNoteView = v.findViewById(R.id.editNoteView);
        editNoteView.setText(mItem.getContent());
        return v;
    }

    @Override
    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
        inflater.inflate(R.menu.menu_edit_note, menu);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.action_save:
                final String newContent = editNoteView.getText().toString();
                new Thread(new Runnable() {
                    @Override
                    public void run() {
                        NotesApplication.getDataManager().updateItem(new Item(mItem.getdAt(), newContent));
                        getActivity().finish();
                    }
                }).run();
                return true;
            case R.id.action_delete:
                deleteDialog();
                return true;
        }
        return super.onOptionsItemSelected(item);
    }

    public static NoteEditFragment newInstance(Item note, boolean VIEW) {
        NoteEditFragment f = new NoteEditFragment();
        if (VIEW) f.setViewOnlyMode();
        Bundle b = new Bundle();
        b.putSerializable("note", note);
        f.setArguments(b);
        return f;
    }

    private void setViewOnlyMode() {
        this.VIEW_FLAG = true;
    }

    protected void deleteDialog() {
        DeleteDialogFragment ddf = DeleteDialogFragment.newInstance(mItem.getdAt().getTime());
        ddf.show(getActivity().getSupportFragmentManager(), "Delete item");
    }
}

File: app/src/main/java/com/me/noteapp/fragment/NoteEditFragment.kt
package com.me.noteapp.fragment


import android.os.AsyncTask
import android.os.Bundle
import android.view.*
import android.widget.EditText
import android.widget.TextView
import androidx.fragment.app.Fragment
import com.me.noteapp.NotesApplication

import com.me.noteapp.R
import com.me.noteapp.entity.Item
import kotlinx.android.synthetic.main.activity_note.view.*
import java.util.*
import kotlin.coroutines.coroutineContext


class NoteEditFragment : Fragment() {
    private lateinit var editNoteView: EditText
    private lateinit var mItem: Item


    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        setHasOptionsMenu(true)
        mItem = arguments?.getSerializable("note") as Item
        val inflatedView = inflater.inflate(R.layout.fragment_note_edit, container, false)
        editNoteView = inflatedView.findViewById<EditText>(R.id.editNoteView)
        editNoteView.setText(mItem.content)
        return inflatedView
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        inflater.inflate(R.menu.menu_edit_note, menu)
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        if (item?.itemId?.equals(R.id.action_save)!!) {
            val newContent = editNoteView.text.toString()
            doAsync {
                NotesApplication.getDataManager().updateItem(Item(mItem.getdAt(), newContent))
                activity?.finish()
            }.execute()
            return true
        }
        return super.onOptionsItemSelected(item)
    }

    class doAsync(val handler: () -> Unit) : AsyncTask<Void, Void, Void>() {
        override fun doInBackground(vararg params: Void?): Void? {
            handler()
            return null
        }
    }

    companion object {
        @JvmStatic
        fun newInstance(note: Item): NoteEditFragment {
            val f = NoteEditFragment()
            val bundle = Bundle()
            bundle.putSerializable("note", note)
            f.arguments = bundle
            return f
        }
    }

}
